code/parachain/frame/assets-transactor-router/src/tests/routing.rs

```diff
@@ -0,0 +1,1092 @@
+			.unwrap();
+
+	let foreign_asset_id = XcmAssetLocation(xcm::v2::MultiLocation::parent());
+	let foreign_asset_info = AssetInfo {
+		name: Some(
+			BiBoundedAssetName::from_vec(b"Kusama".to_vec()).expect("string is within bound"),
+		),
+		symbol: Some(
+			BiBoundedAssetSymbol::from_vec(b"KSM".to_vec()).expect("string is withing bound"),
+		),
+		decimals: Some(12),
+		ratio: None,
+		existential_deposit: 1000,
+	};
+
+	let asset_id_foreign = Pallet::<Test>::create_foreign_asset(
+		protocol_id_foreign,
+		nonce_foreign,
+		foreign_asset_info,
+		foreign_asset_id,
+	)
+	.unwrap();
+	(asset_id_local, asset_id_foreign)
+}
+
+// issue assets to different accounts and in different amount
+fn mint_assets(asset_id_local: AssetId, asset_id_foreign: AssetId) {
+	Pallet::<Test>::mint_into(RuntimeOrigin::root(), asset_id_local, ACCOUNT_LOCAL, 1000).unwrap();
+	Pallet::<Test>::mint_into(RuntimeOrigin::root(), asset_id_foreign, ACCOUNT_FOREIGN, 2000)
+		.unwrap();
+	Pallet::<Test>::mint_into(RuntimeOrigin::root(), NATIVE_ASSET_ID, ACCOUNT_NATIVE, 3000)
+		.unwrap();
+}
+
+// tests for route macro in orml.rs
+mod orml_route {
+	use frame_support::{
+		assert_ok,
+		traits::{Currency, ReservableCurrency},
+	};
+	use orml_traits::{MultiCurrency, MultiReservableCurrency};
+
+	use crate::mocks::{AccountId, RuntimeOrigin};
+
+	use super::*;
+
+	#[test]
+	fn minimum_balance() {
+		// tests ED specified in pallet's Configs via common interface and per pallet
+		new_test_ext().execute_with(|| {
+			let (asset_id_local, asset_id_foreign) = create_assets();
+			assert_eq!(
+				<Pallet::<Test> as MultiCurrency<AccountId>>::minimum_balance(NATIVE_ASSET_ID),
+				1
+			);
+			// should be 0 because of ExistentialDeposits for orml_tokens
+			assert_eq!(
+				<Pallet::<Test> as MultiCurrency<AccountId>>::minimum_balance(asset_id_local),
+				0
+			);
+			// should be 0 because of ExistentialDeposits for orml_tokens
+			assert_eq!(
+				<Pallet::<Test> as MultiCurrency<AccountId>>::minimum_balance(asset_id_foreign),
+				0
+			);
+
+			assert_eq!(<<Test as Config>::NativeTransactor>::minimum_balance(), 1);
+			assert_eq!(<<Test as Config>::LocalTransactor>::minimum_balance(asset_id_local), 0);
+			assert_eq!(<<Test as Config>::ForeignTransactor>::minimum_balance(asset_id_foreign), 0);
+		});
+	}
+
+	#[test]
+	fn total_issuance() {
+		// mint assets and check new issuance via common interface and per pallet
+		new_test_ext().execute_with(|| {
+			let (asset_id_local, asset_id_foreign) = create_assets();
+			mint_assets(asset_id_local, asset_id_foreign);
+			assert_eq!(
+				<Pallet::<Test> as MultiCurrency<AccountId>>::total_issuance(NATIVE_ASSET_ID),
+				3000
+			);
+			assert_eq!(
+				<Pallet::<Test> as MultiCurrency<AccountId>>::total_issuance(asset_id_local),
+				1000
+			);
+			assert_eq!(
+				<Pallet::<Test> as MultiCurrency<AccountId>>::total_issuance(asset_id_foreign),
+				2000
+			);
```

Comment:
as per my comment about constants - they should be used here instead of 1000, 2000, 3000